src/main/java/jep/SubInterpreterOptions.java

```diff
@@ -0,0 +1,195 @@
+     *
+     */
+    public SubInterpreterOptions setAllowFork(boolean allowFork) {
+        this.allowFork = allowFork ? 1 : 0;
+        return this;
+    }
+
+    /**
+     * If this is false then the runtime will not support replacing the current
+     * process via exec (e.g. os.execv()) in any thread where the
+     * sub-interpreter is currently active. Otherwise exec is unrestricted.
+     *
+     * Note that the subprocess module still works when exec is disallowed.
+     *
+     * @param allowExec
+     *            whether the  sub-interpreter will allow exec.
+     * @return a reference to this SubInterpreterOptions
+     *
+     */
+    public SubInterpreterOptions setAllowExec(boolean allowExec) {
+        this.allowExec = allowExec ? 1 : 0;
+        return this;
+    }
+
+    /**
+     * If this is false then the sub-interpreter’s threading module won’t
+     * create threads. Otherwise threads are allowed.
+     *
+     * @param allowThreads
+     *            whether the  sub-interpreter will allow threads.
+     * @return a reference to this SubInterpreterOptions
+     *
+     */
+    public SubInterpreterOptions setAllowThreads(boolean allowThreads) {
+        this.allowThreads = allowThreads ? 1 : 0;
+        return this;
+    }
+
+    /**
+     * If this is false then the sub-interpreter’s threading module won’t
+     * create daemon threads. Otherwise daemon threads are allowed 
+     * (as long as allowThreads is true).
+     *
+     * @param allowDaemonThreads
+     *            whether the  sub-interpreter will allow daemon threads.
+     * @return a reference to this SubInterpreterOptions
+     *
+     */
+    public SubInterpreterOptions setAllowDaemonThreads(boolean allowDaemonThreads) {
+        this.allowDaemonThreads = allowDaemonThreads ? 1 : 0;
+        return this;
+    }
+
+    /**
+     * If this is false then all extension modules may be imported, including
+     * legacy (single-phase init) modules, in any thread where the
+     * sub-interpreter is currently active. Otherwise only multi-phase init
+     * extension modules (see PEP 489) may be imported.
+     *
+     * This must be true if useMainObmalloc is false.
+     *
+     * @param checkMultiInterpeExtensions
+     *            whether the  sub-interpreter will restrict import of legacy modules
+     * @return a reference to this SubInterpreterOptions
+     *
+     */
+    public SubInterpreterOptions setCheckMultiInterpeExtensions(boolean checkMultiInterpeExtensions) {
+        this.checkMultiInterpeExtensions = checkMultiInterpeExtensions ? 1 : 0;
+        return this;
+    }
+
+    /**
+     * This determines the operation of the GIL for the sub-interpreter. When
+     * this is false use (share) the main interpreter’s GIL. When this is true
+     * use the sub-interpreter’s own GIL.
+     *
+     * If this is true then useMainObmalloc must be false.
+     *
+     * @param ownGIL
+     *            whether the  sub-interpreter will use its own GIL.
```

Comment:
double space in between "the  sub-interpreter"

Reply:
Fixed in 4f2db10